projects/step-core/src/lib/modules/basics/components/array-filter/array-filter.component.ts

```diff
@@ -66,10 +66,20 @@ export class ArrayFilterComponent<T = unknown> extends BaseFilterComponent<strin
   }
 
   protected override transformFilterValueToControlValue(value: string): unknown {
+    let result: string[];
+
     if (value.startsWith('(') && value.endsWith(')') && value.includes('|')) {
```

Comment:
@dvladir In addition to the simple FieldFilter based on regex that is used here, the table service now supports so-called CollectionFilter. CollectionFilters are hierarchical filters that support operators like Or, And, etc. Using this to build up the filter passed as payload to the service rest/table/functions would also improve the handling in this function

{"collectionFilter":{"type":"Or", "children": [{"type":"Equals","field":"type","expectedValue":"step.plugins.functions.types.CompositeFunction"},{"type":"Equals","field":"type","expectedValue":"step.plugins.java.GeneralScriptFunction"}]}}

Reply:
@jeromecomte We're already using CollectionFilter for some specific cases, were simple string value or regexp can not be applied.
For example: Date filters on the execution page, scope filter on the parameters page, etc.

`Keyword Type filter` was inherited from `Array Filter`, I guess it was the fastest solution.
But in theory we can change it to use the CollectionFilter instead of regex, as you described it above.